src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -1468,21 +1468,25 @@ private void UpdateWindowProperties(WindowProperties newProperties, bool forceCh
 
                 DwmExtendFrameIntoClientArea(_hwnd, ref margins);
 
-                GetClientRect(_hwnd, out var oldClientRect);
-                var oldClientRectOrigin = new POINT();
-                ClientToScreen(_hwnd, ref oldClientRectOrigin);
-                oldClientRect.Offset(oldClientRectOrigin);
-
-                var newRect = oldClientRect;
 
-                if (newProperties.Decorations == SystemDecorations.Full)
+                if (_shown || forceChanges)
```

Comment:
This line causes #16529. Remove it and the window opens at the correct location.